packages/validator/src/humanizer/formatters/WordingHelper.ts

```diff
@@ -19,10 +19,28 @@ export class WordingHelper {
     }`;
   }
 
-  public static extractPropertyName(pointer: string): string {
-    return pointer
-      .replace(/\/\d+$/, '')
-      .split('/')
-      .pop();
+  public static humanizeTarget(jsonPointer: string): string {
+    const { name, index } = WordingHelper.extractTarget(jsonPointer);
+
+    return index !== null
+      ? `The element at index ${index} in the array \`${name}\``
+      : `The property \`${name}\``;
+  }
+
+  private static extractTarget(jsonPointer: string): {
+    name: string;
+    index: number | null;
+  } {
+    // eslint-disable-next-line @typescript-eslint/typedef,@typescript-eslint/naming-convention
+    const [_dummy, name, index] =
```

Comment:
I would prefer to use a shorter name

```suggestion
    const [_, name, index] =
```